contracts/interchain-token-service/src/contract.rs

```diff
@@ -239,16 +252,49 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
     }
 
     fn deploy_remote_interchain_token(
-        _env: &Env,
-        _caller: Address,
-        _salt: BytesN<32>,
-        _minter: Option<Bytes>,
-        _destination_chain: String,
-        _gas_token: Token,
+        env: &Env,
+        caller: Address,
+        salt: BytesN<32>,
+        destination_chain: String,
+        gas_token: Token,
     ) -> Result<BytesN<32>, ContractError> {
-        // TODO: implementation
+        caller.require_auth();
+
+        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
+        let token_id = Self::interchain_token_id(env, Address::zero(env), deploy_salt);
+        let registered_token_address = Self::token_address(env, token_id.clone());
+        let token = token::Client::new(env, &registered_token_address);
+
+        let token_name = token.name();
+        let token_symbol = token.symbol();
+        let token_decimals = token.decimals();
 
-        todo!()
+        ensure!(
+            token_decimals <= u8::MAX as u32,
```

Comment:
create token metadata from token, and call validate_token_metadata on it instead?